stock_customer_deposit/README.rst

```diff
@@ -0,0 +1,184 @@
+   `More details on development status <https://odoo-community.org/page/development-status>`_
+
+**Table of contents**
+
+.. contents::
+   :local:
+
+Use Cases / Context
+===================
+
+This module was developed because Odoo does not have a customer deposit
+management behaviour.
+
+It will be useful for you if customers place orders and the orders are
+to be invoiced, but you store the stock in your warehouses for delivery
+to the customer at a later date.
+
+Configuration
+=============
+
+To configure this module, you need to:
+
+1. Go to Inventory > Configuration > Settings.
+2. Enable Consignment.
+3. Enable Multi-Step Routes.
+4. Go to Inventory > Configuration > Warehouses.
+5. Select the warehouse where you want to manage customer deposits.
+6. Activate use customer deposits.
+
+Usage
+=====
+
+To use this module, you need to:
+
+Create Customer deposits:
+
+1.  Go to Sales > Quotations.
+2.  Create new quotation to the desired customer.
+3.  Activate **customer deposit** in quotation.
+4.  Add storable product in order lines.
+5.  Add quantity you will keep as a deposit.
+6.  Confirm quotation.
+7.  Two pickings will be created. The first picking will allow
+    simulating the delivery to the customer and being able to invoice
+    the order even if the invoice policy is delivered quantities and the
+    second picking will allow consigning the stock in the same or a
+    different location.
+8.  Validate picking from stock location to Customer.
+9.  Now you could invoice the order
+10. Access the second delivery note, the owner has been automatically
+    assigned.
+11. Select the location where you want to deposit the customer's stock
+    in your warehouse. It can be the same location or a different one.
+
+View Customer deposits:
+
+1. Go to Inventory > Reporting > Locations.
+2. Filter the owner by the customer you created the order.
+3. Go to Sales > Customers.
+4. A smart button with deposits is shown if customer has deposits in
+   some warehouse.
+
+Deliver customer deposits:
+
+1. Go to Sales > Quotation.
+2. Create a new quotation.
+3. Select customer who has a deposit in your warehouse.
+4. In page Other Info choose warehouse where deposit is located. (Only
+   if multi-warehouse is activated)
+5. Smart button **Deposits** with deposits is displayed if the customer
+   has deposits in the chosen warehouse.
+6. Add line with product in deposit.
+7. As a product in deposit, a button **Customer deposit** will appear.
+   If you do not have enough in deposit, button will be grey. If you
+   click on the button **Customer deposit**, you can view the deposits
+   for that product.
+8. You will only be able to confirm the order if you use less quantity
+   than you have in the deposit.
+9. If you try to confirm the order with more quantity than you have in
+   deposit, a validation error will show.
+
+Known issues / Roadmap
+======================
+
+-  The 'qty_at_date_widget' widget that displays the stock on the order
```

Comment:
@EmilioPascual this is not a problem of the widget itself. The problem is everywhere. I mean:

- I buy 1,000 bottles of Valquejigoso v1
- I sell a deposit to Pepephone of 100 bottles of Valquejigoso v1 and validate both pickings
- Go to the product and check "on hand", it is 1,000 no 900
- Make a new sale order to MasMovil of 1,000 bottles, it says is available but it isn't you have 900.
- Go to inventory -> Reporting -> Stock you have 1,000 on hand

It's a know issue but I mean, it is not the widget! the avalabiliy is wrong and reordering rules will not work.